logic.js

```diff
@@ -0,0 +1,142 @@
+let currentPage = 1;
+let totalLimit = 10;
+let totalRecords = 0;
+let totalPages = 0;
+
+const getProducts = async (page, limit, prev = false) => {
+  try {
+    const data = await apiServiceInstance.getProducts(page, limit, prev);
+
+    if (totalRecords !== data.total) {
+      totalRecords = data.total;
+      totalPages = Math.ceil(totalRecords / totalLimit);
+
+      let paginationDiv = document.getElementsByClassName("pageBtn")[0];
+
+      for (let i = 1; i <= totalPages; i++) {
+        let newPage = document.createElement("div");
+        newPage.innerText = i;
+        newPage.setAttribute("class", "pages");
+        newPage.addEventListener("click", (e) => {
+          const newPageNumber = Number(e.target.innerText);
```

Comment:
how many way to convert string to number, read about this...

Reply:
Done